cluster-autoscaler/core/scaleup/resource_manager_test.go

```diff
@@ -0,0 +1,226 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scaleup
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	testprovider "k8s.io/autoscaler/cluster-autoscaler/cloudprovider/test"
+	"k8s.io/autoscaler/cluster-autoscaler/config"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+	"k8s.io/autoscaler/cluster-autoscaler/core/test"
+	"k8s.io/autoscaler/cluster-autoscaler/processors/nodeinfosprovider"
+	kube_util "k8s.io/autoscaler/cluster-autoscaler/utils/kubernetes"
+	utils_test "k8s.io/autoscaler/cluster-autoscaler/utils/test"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+type testCase struct {
```

Comment:
The test cases would be easier to follow if they contained the entire information about the test - including expected results. Having shared `testCase` struct makes this hard. Also, for the same reason (readability), initializing test cases should be done using field names. Otherwise, seeing a line like `{3, 10, 1, 32, 64, 8, 16, 1},`, one has to go back and forth between initialization and struct definition to understand what these numbers represent.
If you really want to keep the common fields in `testCase` for the sake of common initialization code, you can always embed it in unnamed structs instead of relying on matching indices between specific test case and `additionalParams` or `expectedResults`.

Reply:
Agreed, I've refactored all test cases, and they should look better now.